### ISSUE TITLE
New Command to deploy Bicep file to ARM

### ISSUE THREAD

**Issue by PalmEmanuel** [PSBicep/PSBicep#226]
**Is your feature request related to a problem? Please describe.**
After some offline discussion, I'd like to suggest that we implement a command to be able to deploy files directly using PSBicep, instead of creating an implicit dependency for the user by relying on the Azure CLI or PowerShell Module (or Azure portal).

**Describe the solution you'd like**
A new command such as `New-BicepDeployment` with a scope variabel (or several commands based on scope, matching the Az module style).

**Additional context**
The reasoning behind the suggestion is that both the CLI and especially the PowerShell module Az are quite large to download if you're running the deployment from something like a pipeline or container. Only having to install PSBicep would improve the consumer experience, and we have the option of adding functionality such as Build and Restore directly into our Deployment command.

Since we can handle Bicep code as strings, it would also allow us to be able to deploy directly from an online hosting such as the raw file of a GitHub repository without creating a file.

https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/deploy-rest

**Comment by StefanIvemo** [comment]
Should we change the dependency label to `AzAuth`?

**Comment by PalmEmanuel** [comment]
Agreed, implementing AzAuth and making a deployment to the REST API in the module is probably the way we want to proceed with the command.
